### PR TITLE
Fixed #30362 -- Noted partial indexes and constraints restrictions with abstract base classes.

### PR DESCRIPTION
**Pull request by felixxm**
Ticket [30362](https://code.djangoproject.com/ticket/30362).

### CODE REVIEW

File: docs/ref/models/constraints.txt [hunk 1]
```diff
@@ -19,6 +19,16 @@ option.
...
 
+.. admonition:: Constraints in abstract base classes
+
+    You must always specify a unique name for the constraint. This would
+    normally cause a problem in abstract base classes, since the
+    :attr:`Meta.constraints <django.db.models.Options.constraints>` option on
+    this class are included into each of the child classes, with exactly the
+    same values for the attributes (including ``name``) each time. To work
+    around this problem, ``constraints`` option should be specified for each of
+    the child classes with unique names for constraints.
```
Comment (carltongibson) [inline docs/ref/models/constraints.txt]: I'd rephrase without `problem`...

```
    You must always specify a unique name for a constraint. As such, you
    cannot normally specify a constraint on an abstract base class, since the
    :attr:`Meta.constraints <django.db.models.Options.constraints>` option is
    inherited by subclasses, with exactly the same values for the attributes
    (including ``name``) each time. Instead, specify the ``constraints`` option
    on subclasses directly, providing a unique name for each constraint.
```

File: docs/ref/models/indexes.txt [hunk 2]
```diff
@@ -49,6 +49,17 @@ The name of the index. If ``name`` isn't provided Django will auto-generate a
...
+.. admonition:: Partial indexes in abstract base classes
+
+    You must always specify a unique name for the index. This would normally
+    cause a problem for partial indexes in abstract base classes, since
+    ``name`` is required and the
+    :attr:`Meta.indexes <django.db.models.Options.indexes>` option on this
+    class are included into each of the child classes, with exactly the same
+    values for the attributes (including ``name``) each time. To work around
+    this problem, ``indexes`` option should be specified for each of the child
+    classes with unique names for indexes.
```
Comment (carltongibson) [inline docs/ref/models/indexes.txt]: Same as above:

```
    You must always specify a unique name for an index. As such, you
    cannot normally specify an index on an abstract base class, since the
    :attr:`Meta.indexes <django.db.models.Options.indexes>` option is
    inherited by subclasses, with exactly the same values for the attributes
    (including ``name``) each time. Instead, specify the ``indexes`` option
    on subclasses directly, providing a unique name for each index.
```

**Comment by felixxm** [comment]
@carltongibson Thanks for the review :+1:
